tenants-certified-attributes-updater/src/main/scala/it/pagopa/interop/tenantscertifiedattributesupdater/util/package.scala

```diff
@@ -49,30 +49,51 @@ package object util {
     tenants: List[PersistentTenant],
     attributesIndex: Map[UUID, AttributeInfo]
   ): TenantActions = {
-    val fromRegistry: Map[PersistentExternalId, AttributeInfo] =
+    val fromRegistry: List[(PersistentExternalId, AttributeInfo)] =
       institutions
         .filter(_.id.nonEmpty)
         .map { institution =>
           PersistentExternalId(institution.origin, institution.originId) -> AttributeInfo(
             institution.origin,
-            institution.category
+            institution.category,
+            None
           )
         }
-        .toMap
 
     val fromTenant: Map[PersistentExternalId, List[AttributeInfo]] =
-      tenants.map(tenant => tenant.externalId -> tenant.attributes.flatMap(attr => attributesIndex.get(attr.id))).toMap
+      tenants
+        .map(tenant =>
+          tenant.externalId -> tenant.attributes.flatMap(attr =>
+            AttributeInfo.addRevocationTimeStamp(attr, attributesIndex)
+          )
+        )
+        .toMap
 
     val activations: List[InternalTenantSeed] =
       fromRegistry
-        .filterNot { case (extId, attr) => fromTenant.get(extId).exists(_.contains(attr)) }
-        .map { case (extId, attr) =>
-          InternalTenantSeed(ExternalId(extId.origin, extId.value), List(InternalAttributeSeed(attr.origin, attr.code)))
+        .filterNot { case (registryId, attributeFromRegistry) =>
+          fromTenant.get(registryId).exists(AttributeInfo.stillExistsInTenant(attributeFromRegistry))
         }
+        .map { case (externalId, attributeInfo) =>
+          ExternalId(externalId.origin, externalId.value) -> List(
+            InternalAttributeSeed(attributeInfo.origin, attributeInfo.code)
+          )
+        }
+        .groupMapReduce(_._1)(_._2)(_ ++ _)
```

Comment:
```suggestion
        .groupMapReduce[ExternalId, List[InternalAttributeSeed]](_._1)(_._2)(_ ++ _)
```
at least we have a glance of the types